src/components/link/common/link-common.ts

```diff
@@ -0,0 +1,73 @@
+import type { CSSResultGroup, LitElement } from 'lit';
+import { property } from 'lit/decorators.js';
+
+import type { Constructor } from '../../core/common-behaviors';
+import { LanguageController, NamedSlotStateController } from '../../core/common-behaviors';
+import { ACTION_ELEMENTS, hostContext } from '../../core/dom';
+import { actionElementHandlerAspect, HandlerRepository } from '../../core/eventing';
+import type { IsStaticProperty, SbbIconPlacement } from '../../core/interfaces';
+
+import '../../icon';
+
+import style from './link.scss?lit&inline';
+
+export type SbbLinkSize = 'xs' | 's' | 'm';
+
+export declare class SbbLinkCommonInterface {
+  public variant: 'block' | 'inline';
+  public size?: SbbLinkSize;
+  public isStatic: boolean;
+  protected language: LanguageController;
+}
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export const SbbLinkCommonElementMixin = <T extends Constructor<LitElement>>(
+  superClass: T,
+): Constructor<SbbLinkCommonInterface> & T => {
+  class SbbLinkCommonElement
+    extends superClass
+    implements Partial<SbbLinkCommonInterface>, IsStaticProperty
+  {
+    public static styles: CSSResultGroup = style;
+
+    /** Variant of the link (block or inline). */
+    @property({ reflect: true }) public variant: 'block' | 'inline' = 'block';
+
+    /**
+     * Text size, the link should get in the non-button variation.
+     * With inline variant, the text size adapts to where it is used.
+     */
+    @property({ reflect: true }) public size: SbbLinkSize = 's';
+
+    /**
+     * Set this property to true if you want only a visual representation of a
+     * link, but no interaction (a span instead of a link/button will be rendered).
+     */
+    @property({ attribute: 'is-static', reflect: true, type: Boolean }) public isStatic = false;
```

Comment:
Same as with the button. I would prefer a separate component for the static variant.
Depends on discussion of the other comment.

Reply:
Static components created in [fbdfd5b](https://github.com/lyne-design-system/lyne-components/pull/2364/commits/fbdfd5b6821a2323fc942b6aa8ce6538fec1d35a) and mixed static/link behavior fixed in [0edc2f6](https://github.com/lyne-design-system/lyne-components/pull/2364/commits/0edc2f6ec801979752c9132c28709fd89f6c86f3)